src/core/adapters/adapthighlightinfoarray.js

```diff
@@ -0,0 +1,62 @@
+import HighlightedValue from '../models/highlightedvalue';
+
+/**
+ * Constructs a highlighted data object with highlighting applied from an array of
+ * answers-core HighlightInfo models
+ *
+ * @param {HighlightInfo[]} highlightInfoArray from answers-core
+ * @returns {Object<string, string|object>} The object is keyed by fieldName, and it may consist of nested fields
+ *
+ * Example return object:
+ *
+ * {
+ *   name: '<strong>Yext</strong>',
+ *   description: {
+ *     featured: '<strong>Yext</strong> is the offical answers company'
+ *   }
+ * }
+ */
+export default function adaptHighlightInfoArray (highlightInfoArray) {
+  if (highlightInfoArray === undefined || highlightInfoArray.length === 0) {
```

Comment:
Do we want to check if this is falsy since L24 assumes it's truthy?

Reply:
Sure. I'm going to go through these adapters and make my falsy checks a little more robust